lib/http-bridge/test/integration/Cardano/LauncherSpec.hs

```diff
@@ -5,32 +5,54 @@ module Cardano.LauncherSpec
 import Prelude
 
 import Cardano.Launcher
-    ( Command (..), StdStream (..), launch )
-import Control.Concurrent
-    ( threadDelay )
-import Control.Concurrent.Async
-    ( async, cancel, race, wait )
+    ( Command (..), StdStream (..) )
 import Control.Monad
-    ( void )
+    ( when )
+import System.Directory
+    ( doesDirectoryExist, doesFileExist, removePathForcibly )
 import Test.Hspec
-    ( Spec, describe, expectationFailure, it )
+    ( Spec, after_, describe, it )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe )
+import Test.Integration.Framework.DSL
+    ( expectCmdStarts )
 
 spec :: Spec
-spec = describe "cardano-wallet-launcher" $ do
-    it "Can start launcher against testnet" $ do
-        let cardanoWalletLauncher = Command
-                "cardano-wallet-launcher"
-                [ "--http-bridge-port", "8080"
-                ] (return ())
-                Inherit
-        handle <- async $ void $ launch [cardanoWalletLauncher]
-        let fiveSeconds = 5000000
-        winner <- race (threadDelay fiveSeconds) (wait handle)
-        case winner of
-            Left _ -> do
-                cancel handle
-                threadDelay 1000000
-            Right _ ->
-                expectationFailure
-                    "cardano-wallet-launcher isn't supposed to terminate. \
-                    \Something went wrong."
+spec = after_ tearDown $ do
+    describe "LAUNCHER - cardano-wallet-launcher" $ do
+        it "LAUNCHER - Can start launcher against testnet" $ do
+            let cardanoWalletLauncher = Command
+                    "cardano-wallet-launcher"
+                    [ "--http-bridge-port", "8080"
+                    ] (return ())
+                    Inherit
+            expectCmdStarts cardanoWalletLauncher
+            d1 <- doesDirectoryExist stateDir
+            d1 `shouldBe` False
+
+        it "LAUNCHER - Can start launcher against testnet with --state-dir" $ do
+            let cardanoWalletLauncher = Command
+                    "cardano-wallet-launcher"
+                    [ "--http-bridge-port", "8080"
```

Comment:
Depending on which of this one or #414 goes first, we'll have to update this to use the `launch` command instead.